### PR TITLE
Ravinder | Test | ICICI integration

### PR DESCRIPTION
**Pull request by ravinderwalkover**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.html [hunk 1]
```diff
@@ -447,19 +447,286 @@ <h3 class="section-head mb-2">{{localeData?.ecommerce?.shopify_form_heading}}</h
...
+                                            <ng-container *ngFor="let alert of bankAccount?.iciciDetailsResource?.paymentAlerts; let i = index; trackBy: trackByAlertUniqueName;">{{alert?.name}}{{ (i < bankAccount?.iciciDetailsResource?.paymentAlerts?.length - 1) ? ", " : "" }}</ng-container>
+                                        </span>
+                                        <span class="mr-l15 action-div">
+                                            <a href="javascript:;" class="icon-edit-pencil" (click)="openEditAccountModal(bankAccount)"></a>
+                                        </span>
+                                    </div>
+                                </div>    
+                                <div class="font-14">
+                                    <ng-container *ngIf="bankAccount?.iciciDetailsResource?.payor?.length > 0">
+                                        <div class="mt-2" *ngFor="let payor of bankAccount?.iciciDetailsResource?.payor;">
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/settings/integration/setting.integration.component.html:489]: Do use trackBy

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.html [hunk 2]
```diff
@@ -447,19 +447,286 @@ <h3 class="section-head mb-2">{{localeData?.ecommerce?.shopify_form_heading}}</h
...
+                        </div>
+                    </div>
+                    <div class="modal-footer">
+                        <button type="submit" class="btn btn-success mr-1" [disabled]="editAccountUserForm?.invalid">{{commonLocaleData?.app_update}}</button>
+                    </div>
+                </form>
+            </div>
+        </div>
+    </div>
+    <div bsModal #confirmationModal="bs-modal" [config]="{backdrop: 'static', keyboard: false}" class="modal fade"
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/settings/integration/setting.integration.component.html:711]: Consider using `confirm-modal` component

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.ts [hunk 3]
```diff
@@ -347,12 +383,25 @@ export class SettingIntegrationComponent implements OnInit, AfterViewInit {
         this.store.pipe(select(state => state.settings.usersWithCompanyPermissions), takeUntil(this.destroyed$)).subscribe(response => {
             if (response) {
                 this.usersList = [];
+                this.paymentAlersUsersList = [];
+                let index = 0;
+
+                this.paymentAlersUsersList.push({ index: index, label: "Select All", value: this.selectAllRecords });
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/settings/integration/setting.integration.component.ts:389]:
```suggestion
                this.paymentAlertsUsersList.push({ index: index, label: "Select All", value: this.selectAllRecords });
```

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.html [hunk 4]
```diff
@@ -447,19 +447,286 @@ <h3 class="section-head mb-2">{{localeData?.ecommerce?.shopify_form_heading}}</h
...
+                        </div>    
+                    </div>
+
+                    <div class="connected-accounts">
+                        <b class="font-14">{{localeData?.payment?.connected_accounts}}</b><br>
+                        <giddh-page-loader [cssClass]="'mt-0 mb-0'" *ngIf="isLoading"></giddh-page-loader>
+                        <ng-container *ngIf="!isLoading && connectedBankAccounts?.length > 0">
+                            <div class="account-wrapper" *ngFor="let bankAccount of connectedBankAccounts; trackBy: trackByAccountUniqueName;">
+                                <div class="account-info-background width-100 font-14 row">
+                                    <div class="col-md-4">
```
Comment (shalinee-walkover) [inline apps/web-giddh/src/app/settings/integration/setting.integration.component.html:469]: Check responsive as well.

File: apps/web-giddh/src/assets/css/style-2.scss [hunk 5]
```diff
@@ -8496,6 +8496,9 @@ body.update-scroll-hidden {
 .manage-account-popup.full-screen-modal{
     z-index: 999991;
 }
+.v-align-top {
```
Comment (aditi-tiwari17) [inline apps/web-giddh/src/assets/css/style-2.scss:8499]: there is a predefined class="align-top" for vertical-align: top; Please add the class wherever required and remove this from here

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.scss [hunk 6]
```diff
@@ -180,3 +180,92 @@
...
+        color: $grey-chateau;
+    }
+}
+.account-approval-status {
+    margin-left: 13px;
+}
+.field-label,.account-modal label,.user-modal label {
+    color: $omiod;
+}
+.account-approval-status.approved {
```
Comment (shalinee-walkover) [inline apps/web-giddh/src/app/settings/integration/setting.integration.component.scss:250]: CSS should be nested

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.html (hunk 4 above)
Comment (aditi-tiwari17) [inline apps/web-giddh/src/app/settings/integration/setting.integration.component.html:469]: check responsive for this as col-md will work as col-sm in lower screen size

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.ts [hunk 7]
```diff
@@ -1631,6 +1659,349 @@ export class SettingIntegrationComponent implements OnInit, AfterViewInit {
...
+        this.connectedBankAccounts = [];
+
+        this.settingsIntegrationService.getAllBankAccounts().pipe(take(1)).subscribe(response => {
+            this.isLoading = false;
+            if (response?.body) {
+                this.connectedBankAccounts = response.body;
+
+                this.connectedBankAccounts.forEach(bankAccount => {
+                    if(bankAccount?.iciciDetailsResource?.payor?.length > 0) {
+                        bankAccount?.iciciDetailsResource?.payor.forEach(payor => {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/settings/integration/setting.integration.component.ts:1772]: Avoid API calls in for loop

**Review by prateek-giddh — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.html (hunk 4 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/settings/integration/setting.integration.component.html:469]: @shalinee-walkover we don't have to make this responsive.

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.html (hunk 1 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/settings/integration/setting.integration.component.html:489]: Done

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.ts (hunk 3 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/settings/integration/setting.integration.component.ts:389]: Done

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.ts (hunk 7 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/settings/integration/setting.integration.component.ts:1772]: Please ignore as it's approved by Shubhendra sir

File: apps/web-giddh/src/app/settings/integration/setting.integration.component.html (hunk 2 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/app/settings/integration/setting.integration.component.html:711]: Done

File: apps/web-giddh/src/assets/css/style-2.scss (hunk 5 above)
Comment (ravinderwalkover) [reply apps/web-giddh/src/assets/css/style-2.scss:8499]: align-top is for td only, i have added my class name with comma with this.

**Review by aditi-tiwari17 — APPROVED** [review]

**Review by shalinee-walkover — APPROVED** [review]

**Review by shalinee-walkover — APPROVED** [review]

File: apps/web-giddh/src/app/settings/integration/payment/icici/account-create-edit/account-create-edit.component.ts [hunk 8]
```diff
@@ -0,0 +1,266 @@
...
+    }
+
+    /**
+     * This will save the new account
+     *
+     * @memberof AccountCreateEditComponent
+     */
+    public saveNewAccount(): void {
+        if (!this.accountForm?.invalid) {
+            this.accountForm.get('paymentAlerts')?.patchValue(this.paymentAlerts.filter(user => user !== this.selectAllRecords).map(user => user));
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/settings/integration/payment/icici/account-create-edit/account-create-edit.component.ts:155]: Please check `map` could be avoided as it is plainly returning the same data obtained from `filter`

**Review by prateek-giddh — CHANGES_REQUESTED** [review]

**Review by prateek-giddh — APPROVED** [review]

**Pull request merged**